actionpack/test/controller/log_subscriber_test.rb

```diff
@@ -258,6 +259,18 @@ def test_with_fragment_cache
     @controller.config.perform_caching = true
   end
 
+  def test_with_fragment_cache_when_log_disabled
+    @controller.config.perform_caching = true
+    @controller.enable_fragment_cache_logging = false
```

Comment:
We should capture the previous values here and reset them in `ensure`. Otherwise we're potentially leaking state.